db/dcrpg/pgblockchain.go

```diff
@@ -804,6 +817,70 @@ func (pgb *ChainDB) Store(blockData *blockdata.BlockData, msgBlock *wire.MsgBloc
 	return err
 }
 
+// GetTicketsPriceByHeight returns the ticket price by height chart data.
+// This is the default chart that appears at charts page.
+func (pgb *ChainDB) GetTicketsPriceByHeight() (*dbtypes.ChartsData, error) {
+	d, err := RetrieveTicketsPriceByHeight(pgb.db)
+	if err != nil {
+		return nil, fmt.Errorf("RetrieveTicketsPriceByHeight: %v", err)
+	}
+	return &dbtypes.ChartsData{Time: d.Time, ValueF: d.ValueF}, nil
+}
+
+// GetPgChartsData fetches the charts data that is stored in pg
+func (pgb *ChainDB) GetPgChartsData() (map[string]*dbtypes.ChartsData, error) {
+	tickets, err := RetrieveTicketsPriceByHeight(pgb.db)
+	if err != nil {
+		return nil, fmt.Errorf("RetrieveTicketsPriceByHeight: %v", err)
+	}
+
+	supply, err := retrieveCoinSupply(pgb.db)
+	if err != nil {
+		return nil, fmt.Errorf("retrieveCoinSupply: %v", err)
+	}
+
+	size, err := retrieveBlockTicketsPoolValue(pgb.db)
+	if err != nil {
+		return nil, fmt.Errorf("retrieveBlockTicketsPoolValue: %v", err)
+	}
+
+	txRate, err := retrieveTxPerDay(pgb.db)
+	if err != nil {
+		return nil, fmt.Errorf("retrieveTxPerDay: %v", err)
+	}
+
+	ticketsSpendType, err := retrieveTicketSpendTypePerBlock(pgb.db)
+	if err != nil {
+		return nil, fmt.Errorf("retrieveTicketSpendTypePerBlock: %v", err)
+	}
+
+	ticketsByOutputsAllBlocks, err := retrieveTicketByOutputCount(pgb.db, outputCountByAllBlocks)
+	if err != nil {
+		return nil, fmt.Errorf("retrieveTicketByOutputCount by All Blocks: %v", err)
+	}
+
+	ticketsByOutputsTPWindow, err := retrieveTicketByOutputCount(pgb.db, outputCountByTicketPoolWindow)
+	if err != nil {
+		return nil, fmt.Errorf("retrieveTicketByOutputCount by All TP window: %v", err)
+	}
+
```

Comment:
A follow-up PR should create a cache for all of the above queries that do not change until new blocks are mined.

Reply:
The data is cached (in a global variable) already and updated at intervals of five new blocks. The queries take almost a minute to update thus updating after the cache after every new block is mined may not be that effective.